### DISCUSSION TITLE
a small question

### DISCUSSION THREAD

**Discussion by N30Z3N** [montroly/fansyloader#3] [Q&A] [answered]
"Where can I find the source code?
Nowhere. The source code is not public.
Are you planning to publish the source code?
No."
I'm not a coder still just wondering how this actually works

**Comment by montroly** [comment, marked as answer]
Hi @bnsave100,

the program is compiled. So the program code (text files) are converted to an executable file. This makes it not impossible to find out how the program works, but much more difficult. However, you can always check what the program is doing, i.e. what files it is accessing, what network connections it is establishing and what data it is sending/receiving over those connections.

It's the usual procedure if you don't want your program code to become public. Under Windows, compiled programs usually have an .exe extension. 

Most scrapers on Github are written in Python. Python is an interpreted programming language. This means that the program code ( again text files) is read through and executed by another program ( the interpreter). However, in order for the interpreter to read through the program code, it must be publicly viewable (even to the user, i.e. you). (You can work around that, but that's another story that's mostly irrelevant.)

It makes no difference to the user of the application, except that compiled applications require less effort to run. From a security point of view, unless you have read and verified the program code of a public scraper, it is just as much a black box as a compiled scraper. 

This Github repository is only for the documentation and for publishing the executables. 

Maybe this article will help: https://www.freecodecamp.org/news/compiled-versus-interpreted-languages/

**Reply by N30Z3N** [reply]
kk got it :)
